### DISCUSSION TITLE
Is rewriting Uno in MAUI something planned?

### DISCUSSION THREAD

**Discussion by pkar70** [unoplatform/uno#7906] [Ideas]
Maybe answer to question as in Topic is somewhere answered, but I coudn't find it.

It seems like many... ok, not many, but some :) features which are implemented in Uno "by hand" in Uno is covered now by MAUI.
Can we, and whats more, should we make use of it, and replace current separate iOS/macOS/Android implementation by implementation using MAUI?

Of course, it would change minimum platforms versions, e.g.:
Android 6.0 (API 23) would be lowered to Android 5.0 (API 21) or higher,
macOS 10.09 would be raised to macOS 10.13 or higher (using Mac Catalyst).

https://docs.microsoft.com/en-us/dotnet/maui/supported-platforms
https://docs.microsoft.com/en-us/xamarin/cross-platform/get-started/requirements

**Comment by francoistanguay** [comment]
Example? UI or Non-UI? And how would you deal with the fact we also target linux and Wasm?

**Reply by pkar70** [reply]
Of course, targets as WASM should be left as it is.
And yes, both UI and non-UI, e.g. TextBox and Bluetooth.

**Reply by francoistanguay** [reply]
So the short answer is no.
For UI, MAUI doesnt have an implemention of most controls, it's just an abstraction/facade over what's platform-specific and therefore it's way more limited in terms of API surface (intersection of all platforms) than what WinUI offers. So it's not an option.

For Non-UI, you should be able to use Essentials directly if that's the API contract you want to bind to. Again, last time we looked into it, the WinRT (Non UI) apis are way richer than what Essentials offer and the reuse would be extremely limited and not worth the extra dependency.

Trust me, if there's anything worth it we could reuse, we'd be doing it.

**Reply by MartinZikmund** [reply]
Just to add - for some controls like `Grid` , `TreeView`, `NavigationView`, etc. Uno already has the 100 % exact implementation matching UWP/WinUI which is has more features than XF/Maui. Because our goal is to match the UWP/WinUI contract, it is better to reuse the "WinUI" code than to build on something which has different use cases/feature set.
